### PR TITLE
[Hiếu] Chat detail part 2

### PR DESCRIPTION
**Pull request by HieuJC**

### CODE REVIEW

File: app/src/main/java/com/hieuminh/chatapp/presenter/impl/LoginPresenterImpl.kt [hunk 1]
```diff
@@ -25,7 +29,20 @@ class LoginPresenterImpl(private var view: LoginContract.View?) : LoginContract.
...
         mAuth.signInWithCredential(credential)
             .addOnCompleteListener { task ->
-                view?.onUpdateUI(if (task.isSuccessful) mAuth.currentUser else null)
+                if(task.isSuccessful) {
+                    val userProfile = UserProfileModel()
+                    userProfile.id = mAuth.uid
+                    userProfile.email = mAuth.currentUser?.email
+                    userProfile.name = mAuth.currentUser?.displayName
+                    userProfile.avatarUrl = mAuth.currentUser?.photoUrl?.toString()
+                    userProfile.phoneNumber = mAuth.currentUser?.phoneNumber
```
Comment (AnhJobChat) [inline app/src/main/java/com/hieuminh/chatapp/presenter/impl/LoginPresenterImpl.kt]: move into constructor or use apply

**Review by AnhJobChat — APPROVED** [review]

File: app/src/main/java/com/hieuminh/chatapp/presenter/impl/LoginPresenterImpl.kt (hunk 1 above)
Comment (vuphu-jc) [inline app/src/main/java/com/hieuminh/chatapp/presenter/impl/LoginPresenterImpl.kt]: extract single method for `UserProfileModel`

File: app/src/main/java/com/hieuminh/chatapp/presenter/impl/ChatDetailPresenterImpl.kt [hunk 2]
```diff
@@ -13,16 +14,15 @@ class ChatDetailPresenterImpl(private var view: ChatDetailContract.View?) :
     ChatDetailContract.Presenter {
 
     private val mDataBaseReference = FirebaseDatabase.getInstance().reference
-    private val mAuth: FirebaseAuth = FirebaseAuth.getInstance()
-    private var messageModel: MessageModel? = null
+    private var words : MutableList<WordModel>? = null
```
Comment (chuongdevJC) [inline app/src/main/java/com/hieuminh/chatapp/presenter/impl/ChatDetailPresenterImpl.kt]:
```suggestion
    private var mWords : MutableList<WordModel>? = null
```

File: app/src/main/java/com/hieuminh/chatapp/presenter/impl/ChatDetailPresenterImpl.kt [hunk 3]
```diff
@@ -34,13 +34,17 @@ class ChatDetailPresenterImpl(private var view: ChatDetailContract.View?) :
...
     override fun sendMessage(word: WordModel, chatId: String) {
-        messageModel?.messages?.add(0, word)
         val messagesMap = HashMap<String, Any>()
-        messageModel?.messages?.let { messagesMap.put(MESSAGES, it) }
+        if(words == null) {
+            messagesMap[MESSAGES] = listOf(word)
+        } else {
+            words?.add(0, word)
+            messagesMap[MESSAGES] = words!!
+         }
```
Comment (chuongdevJC) [inline app/src/main/java/com/hieuminh/chatapp/presenter/impl/ChatDetailPresenterImpl.kt]: avoid force wrap => !!
let get a trick like that: 
```suggestion
val words = mWords
        if(words == null) {
            messagesMap[MESSAGES] = listOf(word)
        } else {
            words.add(0, word)
            messagesMap[MESSAGES] = words
         }
```

**Review by vuphu-jc — APPROVED** [review]

**Pull request merged**